### ISSUE TITLE
Enforce type=module for ESM

### ISSUE THREAD

**Issue by bmeck** [TheLarkInn/unity-component-specification#3]
Required for export syntax in DOM.

**Comment by TheLarkInn** [comment]
That's a good idea to have. Only thing is that this could technically be an entirely separate module format. So for example just being a different module type could be the signal to V8/Eval/Interpretor that this is an ESM module. 

If that's not strong enough we could certainly talk about having that.

**Comment by ChadKillingsworth** [comment]
If it uses a script tag, we should definitely require `type=module`. I think I am mentioning this 3 times in my Polymer presentation.

**Comment by TheLarkInn** [comment]
That is _if_ it is HTML!

**Comment by bmeck** [comment]
If you aren't writing html please use non-conflicting tag names

**Comment by TheLarkInn** [comment]
Other languages share syntax in one way or another. The goal is to have likeness to respective techniques, that can be parsed accordingly, meanwhile augmenting the behavior. 

Honestly I don't see a problem with script type module. But I also don't see it necessary also.

**Comment by bmeck** [comment]
> Other languages share syntax in one way or another. The goal is to have likeness to respective techniques, that can be parsed accordingly, meanwhile augmenting the behavior.

I agree, as long as we don't ever call this stuff HTML that seems fine.

It seems odd that it looks so much like HTML on that point. Is there a reason to look similar enough to be mistakable but not conform? Doesn't that just add overhead vs making a different syntax?

> Honestly I don't see a problem with script type module. 

Because it invalidates programmer intuition that script tags are Scripts without `type=module`. They now need domain specific knowledge around how their toolchain works w/ pseudo-HTML.

> But I also don't see it necessary also.

I tend to think that things should not be done solely when necessary, very few things are truly necessary. I argue that it is necessary to include `type=module` for consistency if this language which is not HTML is attempting to use HTML semantics without adding to developer fatigue.
